app/cafe.py

```diff
@@ -0,0 +1,28 @@
+import datetime
+
+
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if not visitor.get("vaccine"):
+            raise NotVaccinatedError("you do not have vaccine")
+
+        if (
+                visitor.get("vaccine").get("expiration_date")
+                < datetime.date.today()
+        ):
```

Comment:
Code Efficiency Issue: It's more efficient to store the result of `visitor.get('vaccine')` in a variable before checking its properties. This way, you only need to access the dictionary once.